django/core/checks/security/csrf.py

```diff
@@ -38,3 +40,26 @@ def check_csrf_cookie_secure(app_configs, **kwargs):
         settings.CSRF_COOKIE_SECURE
     )
     return [] if passed_check else [W016]
+
+
+@register(Tags.security)
+def check_csrf_failure_view(app_configs, **kwargs):
+    from django.middleware.csrf import _get_failure_view
+
+    errors = []
+    try:
+        signature = inspect.signature(_get_failure_view())
+        signature.bind(None, reason=None)
+    except ImportError:
+        msg = (
+            "The CSRF failure view '%s' could not be imported." %
+            settings.CSRF_FAILURE_VIEW
+        )
+        errors.append(Error(msg, id='security.E025'))
+    except TypeError:
+        msg = (
+            "The CSRF failure view '%s' does not take the correct number of arguments." %
+            settings.CSRF_FAILURE_VIEW
+        )
+        errors.append(Error(msg, id='security.E024'))
```

Comment:
[limit try clauses](https://adamj.eu/tech/2019/10/02/limit-your-try-clauses-in-python/):

```suggestion
    try:
        view = _get_failure_view()
    except ImportError:
        msg = (
            "The CSRF failure view '%s' could not be imported." %
            settings.CSRF_FAILURE_VIEW
        )
        errors.append(Error(msg, id='security.E025'))
    else:
        try:
            inspect.signature(view).bind(None, reason=None)
        except TypeError:
            msg = (
                "The CSRF failure view '%s' does not take the correct number of arguments." %
                settings.CSRF_FAILURE_VIEW
            )
            errors.append(Error(msg, id='security.E024'))
```

Reply:
Nice suggestion!